gatsby/content/docs/e2ee_cross_signing.mdx

```diff
@@ -0,0 +1,920 @@
+   cross-signing verification. Recursing until you hit a verified device is thus preferred.
+
+## 3. Implementing SSSS
+SSSS is short for Secure Secret Storage and Sharing, which provides a method of storing secret data
+on the server, without the server administrator or anyone else having access to that data. As such,
+this is optimal for storing the private keys for cross-signing on the server so that all your devices
+can use them later on to generate signatures, and thus verify other people.
+
+Unfortunately there doesn't seem to be any help from libolm for implementing SSSS (yet?) so this will
+become rather cryptographic-y to implement. You will mainly need to find libraries for the following
+functions:
+
+1. AES-CTR encryption and decryption
+2. SHA256 HMAC calculation
+3. Base58 encoding and decoding
+4. PBKDF2
+
+As the "sharing" aspect is simpler to implement, here it will be implemented first. It can later on
+also be easily used to debug & verify the functionality of the "secure secret storage" part. It is,
+however, still crucial to look at the general structure of the SSSS first.
+
+TODO: This section doesn't list yet how to *create* keys, add this later on.
+
+### General Structure of SSSS storage
+SSSS depends on keys with which the actual secrets are being encrypted to be stored. These keys need
+to be provided by the client. There is, however, metadata information on the keys in a users account_data.
+For that, there are entries `m.secret_storage.key.<keyId>`, where the keyId is the unique ID of said
+key. In addition, there is an `m.secret_storage.default_key` which denotes the default key to use.
+
+The `m.secret_storage.<keyId>` objects contain a mac and an iv to easily verify if you have the correct
+key, and, in case if there is also a password associated with it, some information on how to generate
+the key out of the password. More specific on that later.
+
+Secrets, such as the private keys for cross-signing, can only be strings (not binary data) and are
+stored encrypted in the account data of your account. For example, the `m.cross_signing.master` entry
+in account data is the private key of your master cross-signing key. Each secret can be encrypted by
+multiple SSS keys. Inside the `encrypted` object there is a mapping of key ID to the ciphertext, iv
+and mac of the secret. A content could look as following:
+
+```json
+{
+  "encrypted": {
+    "awesomeKeyId": {
+      "iv": "supercooliv",
+      "ciphertext": "secretciphertext",
+      "mac": "macthingy"
+    }
+  }
+}
+```
+
+### Caching of secrets
+You have to access some secrets comparatively often. As fetching from the store requires to get a key
+for said secret, that requires user interaction. As such, if a client can store a secret seasonably
+securely client-sided, it is recommended to cache some secrets locally. For cross-signing that would
```

Comment:
I would say something along the lines that each client author should make a decision about which secrets to cache based on tradeoffs between the sensitivity of the information, its ability to keep things secret, and how much of an inconvenience it presents to the user.  As some clients may want to cache more or less than what's recommended.